onnxruntime/core/optimizer/constant_folding.cc

```diff
@@ -20,66 +20,63 @@ Status ConstantFolding::ApplyImpl(Graph& graph, bool& modified, int graph_level)
     if (!node) {
       continue;
     }
-
-    ORT_RETURN_IF_ERROR(Recurse(*node, modified, graph_level));
-
-    InitializedTensorSet constant_inputs;
-
-    // Check if constant folding can be applied on this node.
-    if (!graph_utils::IsSupportedProvider(*node, GetCompatibleExecutionProviders()) ||
-        excluded_op_types_.find(node->OpType()) != excluded_op_types_.end() ||
-        // constant folding does not support executing a node that includes subgraphs (control flow operators,
-        // such as If/Loop/Scan, fall into this category). individual nodes in the subgraph will be processed
-        // by the Recurse call above
-        node->ContainsSubgraph() ||
-        // if the node output is in the graph output, we will get a graph with no nodes.
-        // TODO check if this is allowed in ONNX and ORT.
-        graph.IsNodeOutputsInGraphOutputs(*node) ||
-        !graph_utils::AllNodeInputsAreConstant(graph, *node, constant_inputs)) {
+    try {
+      ORT_RETURN_IF_ERROR(Recurse(*node, modified, graph_level));
+
+      InitializedTensorSet constant_inputs;
+
+      // Check if constant folding can be applied on this node.
+      if (!graph_utils::IsSupportedProvider(*node, GetCompatibleExecutionProviders()) ||
+          excluded_op_types_.find(node->OpType()) != excluded_op_types_.end() ||
+          // constant folding does not support executing a node that includes subgraphs (control flow operators,
+          // such as If/Loop/Scan, fall into this category). individual nodes in the subgraph will be processed
+          // by the Recurse call above
+          node->ContainsSubgraph() ||
+          // if the node output is in the graph output, we will get a graph with no nodes.
+          // TODO check if this is allowed in ONNX and ORT.
+          graph.IsNodeOutputsInGraphOutputs(*node) ||
+          !graph_utils::AllNodeInputsAreConstant(graph, *node, constant_inputs)) {
+        continue;
+      }
+
+      // Create execution frame for executing constant nodes.
+      OptimizerExecutionFrame::Info info({node}, constant_inputs);
+
+      std::vector<int> fetch_mlvalue_idxs;
+      for (auto n : node->OutputDefs()) {
+        if (n->Exists()) {
+          fetch_mlvalue_idxs.push_back(info.GetMLValueIndex(n->Name()));
+        }
+      }
+      std::vector<OrtValue> fetches;
+      Status st = info.RunSingleKernel(fetch_mlvalue_idxs, node->Index(), fetches);
+      if (!st.IsOK())
+        continue;
+
```

Comment:
Is this error logged anywhere? We shouldn't just swallow it as the user now has no idea something failed.